### ISSUE TITLE
From the last update of @keyv/sqlite, suddenly its unable to work properly.

### ISSUE THREAD

**Issue by ShreshthTiwari** [jaredwray/keyv#221]
Connection Error [Error: SQLITE_ERROR: near "907626718095896626": syntax error] {
  errno: 1,
  code: 'SQLITE_ERROR'
}
/home/shreshthtiwari/Documents/GitHub/B0T/node_modules/@keyv/sqlite/src/index.js:44
                        .then(query => query(sqlString));
                                       ^

TypeError: query is not a function
    at /home/shreshthtiwari/Documents/GitHub/B0T/node_modules/@keyv/sqlite/src/index.js:44:19
    at async Client.<anonymous> (/home/shreshthtiwari/Documents/GitHub/B0T/index.js:176:27)

Node.js v17.3.0

**Comment by jaredwray** [comment]
@ShreshthTiwari is this right on when loading node modules or when you are calling it? If so, can you provide any code as reference?

**Comment by jaredwray** [comment]
@ShreshthTiwari one thought on this. Have you validated it with node v16.x.x? That is what we are testing towards at this time.

**Comment by ShreshthTiwari** [comment]
I'm using the latest version 17 maybe.
The code is here 
https://github.com/ShreshthTiwari/B0T/blob/main/builders/databaseBuilder.js

**Comment by ShreshthTiwari** [comment]
And th error is this-
![Screenshot_2022_0131_233722](https://user-images.githubusercontent.com/68736752/151848756-851705ac-d871-476a-b9ba-b092a8bdd572.jpg)

**Comment by jaredwray** [comment]
Are you using NVM (https://github.com/nvm-sh/nvm)? If so, please move it back to v16 and try there. We are looking into this also.

**Comment by ShreshthTiwari** [comment]
Tried in v16 😔
![Uploading Screenshot_2022_0131_234111.jpg…]()

**Comment by ShreshthTiwari** [comment]
![Screenshot_2022_0131_234111](https://user-images.githubusercontent.com/68736752/151849420-53b4999a-1b68-47b6-a0b0-21a5f5ee9d6e.jpg)

**Comment by alphmth** [comment]
@jaredwray we are now using `better-sqlite3`. Was that released or it has been released yet?

**Comment by jaredwray** [comment]
@alphmth - it has been released. can you work to get it working correctly and add in tests to make sure it works?

**Comment by jaredwray** [comment]
@ShreshthTiwari - can you please delete you node_modules and re-install via `npm i` as that line 44 error no longer looks correct.

**Comment by ShreshthTiwari** [comment]
Still 😔

**Comment by ShreshthTiwari** [comment]
![Screenshot_2022_0203_074435](https://user-images.githubusercontent.com/68736752/152270175-e7dcb1b9-f579-49c1-858b-eeba8a8d4c9b.jpg)

**Comment by jaredwray** [comment]
@ShreshthTiwari - I am not able to duplicate this. Do you have a specific code base that we can look at to see what is going on?

**Comment by jaredwray** [comment]
I am wondering if this is something to do with the new module we introduced `better-sqlite3`. Adding in this issue also for reference. 

https://github.com/jaredwray/keyv/commit/93628a89afdbf3d8865e462f5b3ceec8309758f2#commitcomment-66130354

**Comment by jaredwray** [comment]
@ShreshthTiwari - have you gone back to an older version and validated that it works?

**Comment by ShreshthTiwari** [comment]
Yes i moved back to 
"@keyv/sqlite": "^2.0.2"
"keyv": "^4.1.0"
now its working properly
![image](https://user-images.githubusercontent.com/68736752/152712840-94852744-90cf-4cb2-b4e0-ff46aaa5b950.png)

**Comment by ShreshthTiwari** [comment]
node:internal/process/promises:265
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: SQLITE_ERROR: near "940250716893757440": syntax error
Emitted 'error' event on Keyv instance at:
    at KeyvSqlite.<anonymous> (/home/shreshthtiwari/Documents/GitHub/B0T-v13/node_modules/keyv/src/index.js:49:46)
    at KeyvSqlite.emit (node:events:520:28)
    at KeyvSqlite.emit (node:domain:475:12)
    at /home/shreshthtiwari/Documents/GitHub/B0T-v13/node_modules/@keyv/sqlite/src/index.js:41:25
    at async Client.<anonymous> (/home/shreshthtiwari/Documents/GitHub/B0T-v13/events/Application.js:23:25) {
  errno: 1,
  code: 'SQLITE_ERROR'
}

Node.js v17.4.0

**Comment by ShreshthTiwari** [comment]
DO you have discord etc so that i can give you access to the bot's vps

**Comment by ShreshthTiwari** [comment]
Ooh sorry my bad.
It is working.
I had to update keyv to 4.1.1 to make it work.
Don't know why but it wasn't auto updating keyv to 4.1.1

**Comment by ShreshthTiwari** [comment]
Nvm its still showing the issue

**Comment by jaredwray** [comment]
@ShreshthTiwari - Is there a test environment where you can get this to fail? I have tried on multiple linux versions and it is working.

**Comment by jaredwray** [comment]
@ShreshthTiwari what operating system is this on and what node version?

**Comment by ShreshthTiwari** [comment]
OS- Ubuntu version 20
Nodejs version 17

**Comment by ShreshthTiwari** [comment]
> @ShreshthTiwari - Is there a test environment where you can get this to fail? I have tried on multiple linux versions and it is working.

I'm running it on a pterodactyl nodejs docker.

**Comment by ShreshthTiwari** [comment]
/home/shreshthtiwari/Documents/GitHub/bot-new/node_modules/@keyv/sqlite/src/index.js:51
                return this.db.prepare(upsert).run(key, value);
                               ^

TypeError: Cannot read properties of undefined (reading 'prepare')
    at KeyvSqlite.set (/home/shreshthtiwari/Documents/GitHub/bot-new/node_modules/@keyv/sqlite/src/index.js:51:18)
    at /home/shreshthtiwari/Documents/GitHub/bot-new/node_modules/keyv/src/index.js:101:25
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.execute (/home/shreshthtiwari/Documents/GitHub/bot-new/src/events/ready.js:48:5)

Node.js v17.4.0

**Comment by jaredwray** [comment]
@ShreshthTiwari - can you try this on Node v16?

**Comment by ShreshthTiwari** [comment]
Oh its working in 16.4.2 thank you! :)

**Comment by ShreshthTiwari** [comment]
/home/shreshthtiwari/Documents/GitHub/bot-new/node_modules/@keyv/sqlite/src/index.js:38
                const row = this.db.prepare(select).get(key);
                                    ^

TypeError: Cannot read properties of undefined (reading 'prepare')
    at KeyvSqlite.get (/home/shreshthtiwari/Documents/GitHub/bot-new/node_modules/@keyv/sqlite/src/index.js:38:23)
    at /home/shreshthtiwari/Documents/GitHub/bot-new/node_modules/keyv/src/index.js:63:22
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async module.exports (/home/shreshthtiwari/Documents/GitHub/bot-new/src/updater/minecraftServerStatusUpdater.js:5:32)
[nodemon] app crashed - waiting for file changes before starting...

**Comment by ShreshthTiwari** [comment]
![image](https://user-images.githubusercontent.com/68736752/159687441-b0b9b86c-1046-42f2-918a-f4124b9d3b99.png)

**Comment by ShreshthTiwari** [comment]
Hmm i think i found the error.
I think it worked..........

**Comment by ShreshthTiwari** [comment]
node_modules/@keyv/sqlite/src/index.js

instead of ${this.opts.table} it should be "${this.opts.table}"

**Comment by jaredwray** [comment]
@ShreshthTiwari - great. Want to do a pull request so we can validate and merge?

**Comment by ShreshthTiwari** [comment]
> @ShreshthTiwari - great. Want to do a pull request so we can validate and merge?

no it didn't work
the problem is just that table name shouldn't be numeric
i edited the table names to `k + {guild.id}`
and its working now

**Comment by ShreshthTiwari** [comment]
i added a function toString
which will check the first character of tablename
if its not between a-z or A-z
it will put a _ before it 
and it worked :)

**Comment by ShreshthTiwari** [comment]
still a bug
its setting table name as undefined

**Comment by ShreshthTiwari** [comment]
ahh im dumb i didn't return the value from toString function

**Comment by ShreshthTiwari** [comment]
```
'use strict';

const EventEmitter = require('events');
const Database = require('better-sqlite3');
const { Message } = require('discord.js');

function toString(input){
	input = input + '';
	if(!input.charAt(0) !== (/[a-zA-Z]/))
		input = '_' + input;
	return input;
}

class KeyvSqlite extends EventEmitter {
	constructor(options) {
		super();
		this.ttlSupport = false;
		options = Object.assign({
			dialect: 'sqlite',
			uri: 'sqlite://:memory:',
		}, options);
		options.db = options.uri.replace(/^sqlite:\/\//, '');

		this.opts = Object.assign({
			table: 'keyv',
			keySize: 255,
		}, options);

		let tableName = toString(this.opts.table);

		const createTable = `CREATE TABLE IF NOT EXISTS ${tableName}(key VARCHAR(${Number(this.opts.keySize)}) PRIMARY KEY, value TEXT )`;

		const dbOptions = {};
		if (options.busyTimeout) {
			dbOptions.timeout = options.busyTimeout;
		}

		try {
			this.db = new Database(options.db, dbOptions);
			this.db.prepare(createTable).run();
		} catch (error) {
			setImmediate(() => this.emit('error', error));
		}
	}

	get(key) {
		let tableName = toString(this.opts.table);

		const select = `SELECT * FROM ${tableName} WHERE key = ?`;
		const row = this.db.prepare(select).get(key);
		if (row === undefined) {
			return undefined;
		}

		return row.value;
	}

	set(key, value) {
		let tableName = toString(this.opts.table);

		const upsert = `INSERT INTO ${tableName} (key, value)
			VALUES(?, ?) 
			ON CONFLICT(key) 
			DO UPDATE SET value=excluded.value;`;
		return this.db.prepare(upsert).run(key, value);
	}

	delete(key) {
		let tableName = toString(this.opts.table);

		const select = `SELECT * FROM ${tableName} WHERE key = ?`;
		const del = `DELETE FROM ${table} WHERE key = ?`;

		const row = this.db.prepare(select).get(key);
		if (row === undefined) {
			return false;
		}

		this.db.prepare(del).run(key);
		return true;
	}

	clear() {
		let tableName = toString(this.opts.table);

		const del = `DELETE FROM ${tableName} WHERE key LIKE ?`;
		this.db.prepare(del).run(`${this.namespace}:%`);
		return undefined;
	}
}

module.exports = KeyvSqlite;
```

**Comment by ShreshthTiwari** [comment]
all good now no errors :)

**Comment by jaredwray** [comment]
We will have a newer version of sqlite coming in the next week or so that should be based on sqlite3. FYI.

**Comment by jaredwray** [comment]
@ShreshthTiwari - this is being resolved in this pull request: https://github.com/jaredwray/keyv/pull/290

**Comment by jaredwray** [comment]
This should be resolved with the following version `v3.5.0`: https://www.npmjs.com/package/@keyv/sqlite
